src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1956 @@
+    "The composite Clvp model with a text encoder, speech encoder and speech decoder model."
+    "The speech decoder model generates the speech_ids from the text and the text encoder and speech encoder works"
+    "together to filter out the best speech_ids.",
+    CLVP_START_DOCSTRING,
+)
+class ClvpModelForConditionalGeneration(ClvpPreTrainedModel):
+    config_class = ClvpConfig
+
+    def __init__(self, config: ClvpConfig):
+        super().__init__(config)
+
+        if not isinstance(config.text_config, ClvpEncoderConfig):
+            raise ValueError(
+                "config.text_config is expected to be of type ClvpEncoderConfig but is of type"
+                f" {type(config.text_config)}."
+            )
+
+        if not isinstance(config.speech_config, ClvpEncoderConfig):
+            raise ValueError(
+                "config.speech_config is expected to be of type ClvpEncoderConfig but is of type"
+                f" {type(config.speech_config)}."
+            )
+
+        if not isinstance(config.decoder_config, ClvpDecoderConfig):
+            raise ValueError(
+                "config.decoder_config is expected to be of type ClvpDecoderConfig but is of type"
+                f" {type(config.decoder_config)}."
+            )
+
+        self.conditioning_encoder = ClvpConditioningEncoder(config)
+
+        self.speech_decoder_model = ClvpForCausalLM(config)
+
+        self.text_encoder_model = ClvpEncoder(config.text_config)
+        self.speech_encoder_model = ClvpEncoder(config.speech_config)
+
+        self.logit_scale = nn.Parameter(torch.tensor(self.config.logit_scale_init_value))
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_conditioning_encoder_input_embeddings(self):
+        return self.conditioning_encoder.text_token_embedding
+
+    def set_conditioning_encoder_input_embeddings(self, new_embeddings):
+        self.conditioning_encoder.text_token_embedding = new_embeddings
+
+    def get_text_encoder_input_embeddings(self):
+        return self.text_encoder_model.token_embedding
+
+    def set_text_encoder_input_embeddings(self, new_embeddings):
+        self.text_encoder_model.token_embedding = new_embeddings
+
+    def get_speech_encoder_input_embeddings(self):
+        return self.speech_encoder_model.token_embedding
+
+    def set_speech_encoder_input_embeddings(self, new_embeddings):
+        self.speech_encoder_model.token_embedding = new_embeddings
+
+    # taken from the original repo,
+    # link : https://github.com/neonbjb/tortoise-tts/blob/4003544b6ff4b68c09856e04d3eff9da26d023c2/tortoise/api.py#L117
+    def fix_speech_decoder_output(self, autoreg_output: torch.LongTensor) -> torch.LongTensor:
+        """
+        This method modifies the output of the decoder model, such as replacing the `eos_token_id` and changing the
+        last few tokens of each sequence.
+
+        Args:
+            autoreg_output (`torch.LongTensor`):
```

Comment:
Could you align this name with the `transformers` library? Maybe `speech_ids` or `speech_candidates`

Reply:
lets go with `speech_ids`.

Reply:
Lets rename all `speech_candidates` to `speech_ids` to keep it consistent.